spring-integration-xmpp/src/main/java/org/springframework/integration/xmpp/outbound/ChatMessageSendingMessageHandler.java

```diff
@@ -64,26 +85,47 @@ public String getComponentType() {
 	@Override
 	protected void handleMessageInternal(Message<?> message) throws Exception {
 		Assert.isTrue(this.initialized, getComponentName() + "#" + this.getComponentType() + " must be initialized");
-		Object messageBody = message.getPayload();
+		Object payload = message.getPayload();
 		org.jivesoftware.smack.packet.Message xmppMessage = null;
-		if (messageBody instanceof org.jivesoftware.smack.packet.Message) {
-			xmppMessage = (org.jivesoftware.smack.packet.Message) messageBody;
+		if (payload instanceof org.jivesoftware.smack.packet.Message) {
+			xmppMessage = (org.jivesoftware.smack.packet.Message) payload;
 		}
-		else if (messageBody instanceof String) {
+		else {
 			String to = message.getHeaders().get(XmppHeaders.TO, String.class);
 			Assert.state(StringUtils.hasText(to), "The '" + XmppHeaders.TO + "' header must not be null");
 			xmppMessage = new org.jivesoftware.smack.packet.Message(to);
-			if (this.headerMapper != null) {
-				this.headerMapper.fromHeadersToRequest(message.getHeaders(), xmppMessage);
+
+			if (payload instanceof ExtensionElement) {
+				xmppMessage.addExtension((ExtensionElement) payload);
+			}
+			else if (payload instanceof String) {
+				if (this.extensionProvider != null) {
+					String data = (String) payload;
+					if (!xmlPattern.matcher(data).matches()) {
+						data = "<root>" + data + "</root>";
```

Comment:
We should add a comment that this is just an arbitrary tag that is removed by the extension parser.